ga4gh/exceptions.py

```diff
@@ -595,3 +552,25 @@ class RepoManagerException(Exception):
     """
     Signals something went wrong inside the repo manager
     """
+
+
+class DuplicateNameException(RepoManagerException):
+    """
+    The user has tried to insert an object with the same name as
+    an existing object.
+    """
+    def __init__(self, objectName, containerName=None):
+        msg = "Name '{}' already exists".format(objectName)
+        if containerName is not None:
+            msg += " in '{}'".format(containerName)
+        super(DuplicateNameException, self).__init__(msg)
```

Comment:
One of the most interesting parts of the SQL repo is removing the requirement that all containers have unique names. Why are we adding that requirement back in? I believe this is the path to doing away with the in-memory `IdNameMap` objects in `Datasets`. If it is important to maintain that restriction it should be a requirement of the (to be deprecated) datasets module, not the repo.